File: Solutions/9. Palindrome Number/CPP/Palindrome_Number.cpp
#include <iostream>

using namespace std;

class Solution{
public:
    bool isPalindrome(int x){
        string s  = to_string(x);
        int stringLength = s.length();

        for(int i = 0; i < stringLength/2; i++){
            if(s[i] != s[stringLength - 1 - i]){
                return false;
            }
        }

        return true;
    }
};

int main(){
    Solution S;

    //input
    int x = 12321;

    cout << S.isPalindrome(x) << endl;

    return 0;
}

File: Solutions/9. Palindrome Number/Python/Palindrome_Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        s = str(x)
        stringLength = len(s)

        for i in range(int(stringLength/2)):
            if s[i] != s[-1 - i]:
                return False

        return True

S = Solution()

print(S.isPalindrome(123))